Fix container_system linker errors in database_system build

## Summary

When building database_system, linker errors occur for container_system symbols.

## Error Details

```
Undefined symbols for architecture arm64:
  "container_module::value_container::set_message_type(std::__1::basic_string_view<char, ...>)"
  "container_module::value_container::to_json()"
  "container_module::value_container::value_container(std::__1::vector<unsigned char, ...> const&, bool)"
  ...
```

## Affected Component

- `database/protocol/database_protocol_container.cpp`
- `tests/integration/integrated_container_protocol_test.cpp`

## Root Cause

The CMake configuration may not be properly linking the container_system library to targets that use `value_container`.

## Suggested Fix

1. Review `database/CMakeLists.txt` to ensure container_system is linked properly
2. Check if container_system build produces the required symbols
3. Verify include paths and library paths are correct

## Related

This issue was discovered during #239 (Result<T> migration) but is unrelated to those changes.

--- a/database/CMakeLists.txt
+++ b/database/CMakeLists.txt
@@ -379,6 +379,25 @@ if(USE_CONTAINER_SYSTEM)
 
             if(_DATABASE_CONTAINER_TARGET)
                 target_link_libraries(${PROJECT_NAME} PUBLIC ${_DATABASE_CONTAINER_TARGET})
+                message(STATUS "Database: Linked container_system via target: ${_DATABASE_CONTAINER_TARGET}")
+            else()
+                # Fallback: find and link library file directly when no target exists
+                find_system_dependency_library(container_system)
+                if(container_system_LIBRARY_DIR)
+                    find_library(_CONTAINER_LIB
+                        NAMES container_system
+                        PATHS "${container_system_LIBRARY_DIR}"
+                        NO_DEFAULT_PATH
+                    )
+                    if(_CONTAINER_LIB)
+                        target_link_libraries(${PROJECT_NAME} PUBLIC ${_CONTAINER_LIB})
+                        message(STATUS "Database: Linked container_system library directly: ${_CONTAINER_LIB}")
+                    else()
+                        message(WARNING "Database: container_system library file not found in ${container_system_LIBRARY_DIR}")
+                    endif()
+                else()
+                    message(WARNING "Database: container_system library directory not found - linking may fail")
+                endif()
             endif()
         else()
             message(WARNING "Database: container_system headers not found - integration disabled")

